Dịch "bert_vn" - Phần 3 (#2554)
aivivn/d2l-vn: chapter_natural-language-processing-pretraining/bert_vn.md

conceptually simple yet empirically powerful pretraining of deep representations for natural languages have revolutionized solutions to various natural language processing tasks.
-->

> BERT cải thiện xa hơn kỹ thuật gần đây nhất đối với mười một tác vụ xử lý ngôn ngữ tự nhiên ở một số hạng mục trải rộng gồm i) phân loại văn bản đơn (cụ thể, phân tích cảm xúc), ii) phân loại cặp văn bản (cụ thể, suy diễn ngôn ngữ tự nhiên), iii) trả lời câu hỏi, iv) đánh tag văn bản (cụ thể, nhận dạng tên riêng).

Reviewer: ```suggestion
BERT cải thiện tốt hơn kết quả tân tiến nhất đối với mười một tác vụ xử lý ngôn ngữ tự nhiên ở phần rộng hạng mục gồm i) phân loại văn bản đơn (cụ thể, phân tích cảm xúc), ii) phân loại cặp văn bản (cụ thể, suy diễn ngôn ngữ tự nhiên), iii) trả lời câu hỏi, iv) đánh tag văn bản (cụ thể, nhận dạng thực thể có tên).
```

`tag` mình có nên dịch ra là `thẻ` không

> BERT cải thiện tốt hơn kết quả tân tiến nhất đối với mười một tác vụ xử lý ngôn ngữ tự nhiên ở phần rộng hạng mục gồm i) phân loại văn bản đơn (cụ thể, phân tích cảm xúc), ii) phân loại cặp văn bản (cụ thể, suy diễn ngôn ngữ tự nhiên), iii) trả lời câu hỏi, iv) đánh tag văn bản (cụ thể, nhận dạng thực thể có tên).

Reviewer 2: ```suggestion
BERT cải thiện tốt hơn kết quả tân tiến nhất đối với mười một tác vụ xử lý ngôn ngữ tự nhiên ở phần rộng hạng mục gồm i) phân loại văn bản đơn (cụ thể, phân tích cảm xúc), ii) phân loại cặp văn bản (cụ thể, suy diễn ngôn ngữ tự nhiên), iii) trả lời câu hỏi, iv) đánh tag văn bản (cụ thể, nhận dạng thực thể có tên).
```
#2637 cho `text tagging`.

> Tất cả các kỹ thuật được đề xuất trong năm 2018, từ ELMO nhạy ngữ cảnh cho tới GPT không phân biệt tác vụ và BERT, đều có ý tưởng đơn giản nhưng thông qua tiền huấn luyện hiệu quả có tính thực nghiệm các biểu diễn sâu cho ngôn ngữ tự nhiên đã tạo ra những giải pháp cách mạng đối với các nhiệm vụ đa dạng của xử lý ngôn ngữ tự nhiên.

Reviewer: ```suggestion
Tất cả các kỹ thuật được đề xuất trong năm 2018, từ ELMO nhạy ngữ cảnh cho tới GPT không phân biệt tác vụ và BERT, đều có ý tưởng đơn giản nhưng là những phương pháp tiền huấn luyện hiệu quả trên thực nghiệm cho các biểu diễn sâu của ngôn ngữ tự nhiên, đã tạo ra những giải pháp cách mạng đối với các tác vụ đa dạng của xử lý ngôn ngữ tự nhiên.
```

> Tất cả các kỹ thuật được đề xuất trong năm 2018, từ ELMO nhạy ngữ cảnh cho tới GPT không phân biệt tác vụ và BERT, đều có ý tưởng đơn giản nhưng là những phương pháp tiền huấn luyện hiệu quả trên thực nghiệm cho các biểu diễn sâu của ngôn ngữ tự nhiên, đã tạo ra những giải pháp cách mạng đối với các tác vụ đa dạng của xử lý ngôn ngữ tự nhiên.

Reviewer 2: ```suggestion
Tất cả các kỹ thuật được đề xuất trong năm 2018, từ ELMo nhạy ngữ cảnh cho tới GPT không phân biệt tác vụ và BERT, đều có ý tưởng đơn giản nhưng là những phương pháp tiền huấn luyện hiệu quả trên thực nghiệm cho các biểu diễn sâu của ngôn ngữ tự nhiên, đã tạo ra những giải pháp cách mạng đối với các tác vụ đa dạng của xử lý ngôn ngữ tự nhiên.
```

<!--
In the rest of this chapter, we will dive into the pretraining of BERT.
When natural language processing applications are explained in :numref:`chap_nlp_app`,
we will illustrate fine-tuning of BERT for downstream applications.
-->

Ở phần còn lại của chương này, ta sẽ đào sâu vào tiền huấn luyện BERT.
Khi những ứng dụng xử lý ngôn ngữ tự nhiên được giải thích trong :numref:`chap_nlp_app`,

> ta sẽ minh họa việc tinh chỉnh BERT cho các ứng dụng xuôi dòng. 

Reviewer: ```suggestion
ta sẽ minh họa việc tinh chỉnh BERT cho các ứng dụng hạ nguồn. 
```

Author: chữ hạ nguồn anh thấy nó tối nghĩa

Reviewer: Anh @davidnvq đề xuất cách dịch này ở https://github.com/aivivn/d2l-vn/pull/2537
Em nghĩ nếu chưa ổn thì mọi người có thể đề xuất lại

Author: Ok, anh cũng đang chờ mọi người thống nhất :)

Reviewer 3: mình lựa chọn xuôi dòng mn nhé.

For instance, one *BERT input sequence* may include either one *text sequence* or two *text sequences*.
-->

> Trong xử lý ngôn ngữ tự nhiên, một số nhiệm vụ (cụ thể, phân tích cảm xúc) lấy một câu văn làm đầu vào, 

Reviewer: ```suggestion
Trong xử lý ngôn ngữ tự nhiên, một số táv vụ (cụ thể, phân tích cảm xúc) lấy một văn bản làm đầu vào, 
```

Author: Chỗ này họ dùng chữ "single text" nên ám chỉ là một câu văn

Reviewer: Em đồng ý.

> trong khi một số nhiệm vụ khác (cụ thể, suy diễn ngôn ngữ tự nhiên), đầu vào là một cặp chuỗi văn bản.

Reviewer: ```suggestion
trong khi một số tác vụ khác (cụ thể, suy diễn ngôn ngữ tự nhiên), đầu vào là một cặp chuỗi văn bản.
```

> Chuỗi đầu vào BERT biểu diễn một cách tường minh cả câu văn và cặp văn bản.

Reviewer: ```suggestion
Chuỗi đầu vào BERT biểu diễn một cách tường minh cả văn bản đơn và văn bản kép.
```

Author: Từ "kép" là hai câu. Nhưng từ "cặp" lại là tương ứng một-một.

> Ở nhiệm vụ đầu, chuỗi đầu vào BERT việc ghép các token của chuỗi văn bản đầu “&lt;cls&gt;”, chuỗi văn bản thứ hai “&lt;sep&gt;” và chuỗi “&lt;sep&gt;”.

Reviewer: ```suggestion
Ở tác vụ đầu, chuỗi đầu vào BERT là sự ghép nói ghép nối của token phân loại đặc biệt “&lt;cls&gt;”, token của chuỗi văn bản, và token phân tách đặc biệt “&lt;sep&gt;”.
Ở tác vụ thứ hai, chuỗi đầu vào BERT là sự ghép nối của “&lt;cls&gt;”, token của chuỗi văn bản đầu, “&lt;sep&gt;”, token của chuỗi văn bản thứ hai, và “&lt;sep&gt;”.
```

For instance, one *BERT input sequence* may include either one *text sequence* or two *text sequences*.
-->

Trong xử lý ngôn ngữ tự nhiên, một số nhiệm vụ (cụ thể, phân tích cảm xúc) lấy một câu văn làm đầu vào, 
trong khi một số tác vụ khác (cụ thể, suy diễn ngôn ngữ tự nhiên), đầu vào là một cặp chuỗi văn bản.
Chuỗi đầu vào BERT biểu diễn một cách tường minh cả văn bản đơn và văn bản kép.
Ở tác vụ đầu, chuỗi đầu vào BERT là sự ghép nói ghép nối của token phân loại đặc biệt “&lt;cls&gt;”, token của chuỗi văn bản, và token phân tách đặc biệt “&lt;sep&gt;”.
Ở tác vụ thứ hai, chuỗi đầu vào BERT là sự ghép nối của “&lt;cls&gt;”, token của chuỗi văn bản đầu, “&lt;sep&gt;”, token của chuỗi văn bản thứ hai, và “&lt;sep&gt;”.
Ta sẽ phân biệt nhất quán thuật ngữ "chuỗi đầu vào BERT" khác với các kiểu "chuỗi" khác.

> Chẳng hạn, một *chuỗi đầu vào BERT* có thể bao gồm cả một *chuỗi văn bản* hoặc hai *chuỗi văn bản*.

Reviewer 2: ```suggestion
Chẳng hạn, một *chuỗi đầu vào BERT* có thể bao gồm cả *một chuỗi văn bản* hoặc *hai chuỗi văn bản*.
```
em thấy in nghiêng cả số lượng hợp lý hơn.

For single text inputs, only $\mathbf{e}_A$ is used.
-->

Để phân biệt cặp văn bản, các embedding phần đã học $\mathbf{e}_A$ and $\mathbf{e}_B$ được thêm vào lần lượt các embeddings token của chuỗi thứ nhất và chuỗi thứ hai.

> Đối với đầu vào là câu văn, ta chỉ sử dụng $\mathbf{e}_A$.

Reviewer: ```suggestion
Đối với đầu vào là văn bản đơn, ta chỉ sử dụng $\mathbf{e}_A$.
```

For single text inputs, only $\mathbf{e}_A$ is used.
-->

> Để phân biệt cặp văn bản, các embedding phần đã học $\mathbf{e}_A$ and $\mathbf{e}_B$ được thêm vào lần lượt các embeddings token của chuỗi thứ nhất và chuỗi thứ hai.

Reviewer 2: ```suggestion
Để phân biệt cặp văn bản, các embedding đoạn đã học $\mathbf{e}_A$ and $\mathbf{e}_B$ được thêm vào lần lượt các embedding token của chuỗi thứ nhất và chuỗi thứ hai.
```

Đối với đầu vào là văn bản đơn, ta chỉ sử dụng $\mathbf{e}_A$.

<!--
The following `get_tokens_and_segments` takes either one sentence or two sentences as the input, 
then returns tokens of the BERT input sequence and their corresponding segment IDs.
-->

Hàm `get_tokens_and_segments` sau đây có thể lấy một hoặc hai câu làm đầu vào,

> rồi trả về các token của chuỗi đầu vào BERT và các ID phần tương ứng của chúng.

Reviewer 2: ```suggestion
rồi trả về các token của chuỗi đầu vào BERT và các ID đoạn tương ứng của chúng.
```
